src/csharp/Microsoft.Spark/Sql/Types/ComplexTypes.cs

```diff
@@ -79,6 +92,19 @@ public sealed class MapType : DataType
         /// Initializes the <see cref="MapType"/> instance.
         /// </summary>
         public MapType() => throw new NotImplementedException();
+
+        /// <summary>
+        /// Does this type need to convert between C# object and internal SQL object.
+        /// </summary>
+        /// <returns>True if conversion needed, false otherwise</returns>
+        internal override bool NeedConversion() => true;
```

Comment:
why do you have two interfaces where there is dependencies instead of having just one interface?

Reply:
Are you referring to having both `NeedConversion` and `FromInternal`?

Reply:
Removed `NeedConversion`

Reply:
Removed both.  Reverted back to original implementation.